File: chrestomathy_code/tasks/math/factorial/factorial_rec.swift
import Foundation

func factorial(_ n: Int) -> Int {
    if n <= 1 {return 1}
    return n * factorial(n-1)
}

let n : Int! = Int(CommandLine.arguments[1])

if n < 0 {
    print("_invalid_")
    exit(0)
}

print(factorial(n))

File: chrestomathy_code/tasks/math/factorial/factorial_rec.c
#include <stdio.h>
#include <stdlib.h>

unsigned long factorial (unsigned long n) {
    return (n < 2) ? 1 : (n * factorial(n - 1));
}

int main (int argc, char *argv[]) {
    int n = atoi(argv[1]);
    if (n < 0) {
        printf("_invalid_");
        exit(0);
    }
    printf("%ld", factorial(n));
    return 0;
}
